### DISCUSSION TITLE
Error while issuing Vcs

### DISCUSSION THREAD

**Discussion by SethiShivam** [walt-id/.github#17] [Help] [answered]
Hi team , 

I am trying to run ssi kit examples

When ever I try to execute this piece of code  to issue  credentials . 

    **public ProofConfig createProofConfig(String issuerDid, String subjectDid, ProofType proofType, Instant expiration) {
    	
        return new ProofConfig(issuerDid, subjectDid, null, null, proofType, null, null,
                null, null, null, null, expiration, null, null, null, null);
    }**

**I get the following error** :

java.lang.NullPointerException: Parameter specified as non-null is null: method id.walt.signatory.ProofConfig.<init>, parameter ecosystem
	at id.walt.signatory.ProofConfig.<init>(Signatory.kt)

But I have not passed any null parameter 

below is the list of parameters I am passing 

ISSUER DID : 
did:ebsi:zkDh3Tw2wNr4tofy62SYHBC
  subject DID : 
did:key:z6MkicPLVe5bX6mEG6kLDcXftpByLh3pqdgWHAgE8b5wypds
 PROOF TYPE 
LD_PROOF
 EXPIRATION 
2022-12-11T10:10:59.389344300Z

But I have checked that there is a last parameter in the createProofConfig()     [Ecosystem Ecosystem]      which should not be null but getting passed as null .

Best Regards
Sethi shivam

**Comment by SethiShivam** [comment]
@taminobaumann @philpotisk

**Comment by philpotisk** [comment]
Hi @SethiShivam thanks for pointing this out. As you have suggested correctly, the last parameter for indicating the ecosystem is not optional any more. I have fixed this in the example project now https://github.com/walt-id/waltid-ssikit-examples/commit/dd1a26487105a717a8b28f71b9ccf288f95138f0

**Comment by SethiShivam** [comment]
Hi @philpotisk  thanks for the response ,

i want to learn the use of ecosystem variable , 
Could you please help me with this , why do we need this "Ecosystem"

or what is it representing ?

Best Regards
Sethi Shivam

**Comment by philpotisk** [comment, marked as answer]
Hi @SethiShivam good question. We realized that the various ecosystems have certain requirements when it comes to issuing and verifying credentials that's why we introduced this parameter for being able to react accordingly. We are just building up the corresponding docs here https://docs.walt.id/v/ssikit/ecosystems/, but have not completed everything yet. The next ecosystems that we are about to introduce are Velocity & Gaia-X, so stay tuned ;-)
